Last Documents added

RojoRX opened the pull request:
Agregue el componente para mostrar los últimos documentos publicados, no logro encontrar la url de la api que necesito para recibir estos datos, por lo que uso de momento otra api para probar que lo demás funciona correctamente

Commit:
Last Documents added

Ditmar commented on src/components/LastDocsComponent/LastDocs.jsx:
```diff
@@ -0,0 +1,77 @@
+        
+    }
+    componentDidMount() {
+        fetch('https://fakestoreapi.com/products?limit=5').then(res => res.json()).then(json => {
```
componentDidMount is using in react with classes, that is, react legacy, in this case, we are using react with a hook if you need to control the fetch you need to put the request, into the useEffect function,

Ditmar commented on src/components/DashboardComponent/Orders.jsx:
```diff
@@ -1,86 +1,78 @@
+            </Typography>
+            <Table size="small" sx={{mt:2, ml:0}}>
+                <TableBody> {
+                    this.state.items.map((row) => (
```
Please try to separate this large conde in small components

Ditmar reviewed (changes requested):
Please use useEffect the fetch function should be in the async directory; use the functions created in https://github.com/Ditmar/plagiarist-g2/blob/master/src/async/api.js to make the request

Commit:
first pull request corrected

RojoRX commented:
Good evening Engineer. I corrected the code problems, now using useEffect and making the code shorter with other components

Ditmar commented on src/components/DashboardComponent/Orders.jsx:
```diff
@@ -1,86 +1,78 @@
+    event.preventDefault();
 }
+class LastDocs extends React.Component {
+    constructor(props) {
```
We are using react with hooks , no React with classes

Ditmar reviewed (changes requested):
don't use classes use hooks

Commit:
Second PR corrected

RojoRX commented:
Good afternoon engineer. Fixed the changes, forgot I was using Orders.jsx as a test at the start. Now all my code is in the LastDocsComponent with the requested changes and I removed my code from Orders.jsx

Ditmar commented on src/components/LastDocsComponent/services/call.js:
```diff
@@ -0,0 +1,5 @@
+const apiData = async (state) => {
+    const response = await fetch("https://fakestoreapi.com/products?limit=5");
+    state(await response.json())
```
nice

Ditmar reviewed (approved):
approved
